### ISSUE TITLE
fetchRequestToken fix $request->setContent(''); breaks some Wordpress Oauth1.0a

### ISSUE THREAD

**Issue by drsdre** [yiisoft/yii2-authclient#152]
### What steps will reproduce the problem?
Using the extension https://github.com/drsdre/yii2-wordpress-api and [Wordpress oAuth1.0a plugin](https://wordpress.org/plugins/rest-api-oauth1/). Setting up an oAuth1.0a authentication request to a Wordpress site using the following code in the controller:
```
// Build website URL
$api_url = $this->model->website->hostname;

// Open Wordpress Auth object
$oauthClient = new OAuth1([
	'apiBaseUrl' => $api_url,
	'consumerKey' => $this->model->client_key,
	'consumerSecret' => $this->model->client_secret,
]);

try {
	if (is_null($oauth_token)) {
		// If no authorisation token, start authorization web flow

		// Must set return URL without parameter to prevent 'OAuth signature does not match' error
		$oauthClient->setReturnUrl(
			yii::$app->getRequest()->getHostInfo().'/'.
			yii::$app->getRequest()->getPathInfo().'?id='.$id);

		// Get request token
		$oauth_token = $oauthClient->fetchRequestToken();

		// Get authorization URL
		$url         = $oauthClient->buildAuthUrl($oauth_token);

		// Redirect to authorization URL
		return $this->redirect($url);
	}

	// After user returns at our site:
	$access_token = $oauthClient->fetchAccessToken($oauth_token);
	$this->model->access_token = yii\helpers\Json::encode($access_token->params); // Upgrade to access token
	$result = $this->model->save();
} catch (yii\base\Exception $e) {
	yii::$app->session->setFlash( 'alert', [
		'body'    => yii::t( 'app', 'Verification failed. Error: ' ).$e->getMessage(),
		'options' => [ 'class' => 'alert-danger' ],
	] );
}
```

### What's expected?
Get an authentication flow started which provides a authorize box on the Wordpress site.

### What do you get instead?
On some sites the Wordpress site sends back an error "400, message: No OAuth parameters supplied". When $request->setContent(''); in function fetchRequestToken is commented out, the flow works as intended. The content is set to null before this line.

Note: this does not occur on all Wordpress sites. Some will not give the error message when $request->setContent(''); is NOT commented out.

There is a possibility that the Wordpress oAuth1.0 plugin is causing this behavior. 

### Additional info

Screenshot failed request before $request->setContent('');
![wordpress_oauth1 0a_request_fail](https://cloud.githubusercontent.com/assets/809827/23458926/73919672-fe7e-11e6-854b-f9ee507c41dd.jpg)

Screenshot working oAuth1.0 request before $request->setContent('');
![wordpress_oauth1 0a_request_works](https://cloud.githubusercontent.com/assets/809827/23458947/84907c04-fe7e-11e6-92b2-5289a6ff6e7f.jpg)

| Q                | A
| ---------------- | ---
| Yii version | 2.0.11.2 |
| yii2-authclient | 2.1.2 |
| yii2-httpclient | 2.0.3 |
| PHP version | 7.0 |
| Operating system | Linux Ubuntu 14.04 |

**Comment by drsdre** [comment]
After some more investigation, it looks like the flow does work when content is set to null i.e.
`$request->setContent(null);`

**Comment by klimov-paul** [comment]
> When $request->setContent(''); in function fetchRequestToken is commented out, the flow works as intended.

This line can not be removed, see #135

> There is a possibility that the Wordpress oAuth1.0 plugin is causing this behavior.

That is most likely.

**Comment by drsdre** [comment]
Please read my last comment. This line should not be removed but the content should be set to null instead of empty string. This changes the content of the oAuth request in such a way that it will work in more situations.

See the prepare function in the Request.php file:

```
    /**
     * Prepares this request instance for sending.
     * This method should be invoked by transport before sending a request.
     * Do not call this method unless you know what you are doing.
     * @return $this self reference.
     */
    public function prepare()
    {
        $content = $this->getContent();
        if ($content === null) {     <-- 
            $this->getFormatter()->format($this);
        } elseif (is_array($content)) {
            $this->prepareMultiPartContent($content);
        }
        $this->isPrepared = true;
        return $this;
    }
```

**Comment by harrytang** [comment]
Same problem here with my FlickAuth.php
Works fine when comment out `$request->setContent('');` or change it to  `$request->setContent(null);`

**Comment by klimov-paul** [comment]
According to [RFC](https://tools.ietf.org/html/rfc5849#section-2.3) `oauth` parameters should not be duplicated. Once they appear at `Authorization` header they should not be passed at request params.

However issue may appear in case `request token` request is performed without `Authorization` header composition.

Resolved by commit 9402529b3576b7f051713e7c6670e9579d358705
